lib/galaxy/managers/workflows.py

```diff
@@ -475,7 +475,7 @@ def _workflow_to_dict_editor(self, trans, stored):
             # Get user annotation.
             annotation_str = self.get_item_annotation_str(trans.sa_session, trans.user, step) or ''
             config_form = None
-            if trans.history:
+            if trans.history or trans.workflow_building_mode == workflow_building_modes.ENABLED:
```

Comment:
This if-statement should probably be just removed. The config form should always be provided when requesting a dictionary for the workflow editor.

Reply:
Per the comment in the statement, we don't want the config form if it isn't a web request.  I assume this was at some point an intentional API optimization -- @jmchilton?

Reply:
I'm plus 1 on dropping the if, I doubt it was an optimization - I just don't think we should default to serving up config forms via the API by default (e.g. to bioblend). They are unstable and they shouldn't be consumed external to the UI client. I will say the correct way to handle that is probably not with ifs like this though - we need separate APIs for things only interesting to the client and not general purpose API consumption and short of that some sort of flag indicating the requestor - that is why I'm totally fine with dropping the if for this case.